libdnf/dnf-context.cpp

```diff
@@ -332,6 +334,8 @@ dnf_context_init(DnfContext *context)
 {
     DnfContextPrivate *priv = GET_PRIVATE(context);
 
+    libdnf::Log::setLogger(&glibLogger);
```

Comment:
Unless I've missed it, microdnf doesn't log at all now.

The messages logged by libdnf are expected to be logged into a log file, not to the user-facing output. The default glog handler logs the messages to sdout/stderr. While most messages in libdnf are logged at DEBUG (for many of those the log level is wrong and we may need to get around to fixing them), which isn't logged by the default glog handler, I don't think the logger messages should suddenly start showing up in either microdnf or any other context interface user's output...

I think it would be better to add a logger setup function to the context interface that will call this `libdnf::Log::setLogger(&glibLogger);` and also allow the caller to either set logfile or potentially log to stdout/stderr. But yeah, this lead to microdnf having its own log file and us having to deal with rotation etc.?

Reply:
Or, come think of it, the specification of the log file is up to the API user who can set up the glog handler they want. The point is if they don't do that, the default handler will start logging to stdout/stderr. So I'm not sure how to do this, maybe just a simple function that will make libdnf log to glog, but optionally, not by default?

Reply:
@lukash 
PackageKit logs to the system logger.
Microdnf does not set the logger - messages go to stdout/stderr. This is nothing new.

The "context" part of libdnf uses the GLib logger from the beginning. This PR adds only a few messages from the new code (which uses the C++ logger, e.g. modularity code).
In other words, microdnf printed log messages on stdou/stderr even before this PR. In addition, when I tested microdnf, only important messages (not debug messages) were printed. In most cases, it is not printed more log messages than before this PR.

Reply:
@lukash
> While most messages in libdnf are logged at DEBUG (for many of those the log level is wrong and we may need to get around to fixing them), which isn't logged by the default glog handler, ...

Yes, microdnf does not print DEBUG messages by default. But printing of these messages can be enabled by environment variable G_MESSAGES_DEBUG.
In GLib documentation:
G_MESSAGES_DEBUG: A space-separated list of log domains for which debug and informational messages are printed. By default these messages are not printed.

Examples:
```
G_MESSAGES_DEBUG=libdnf microdnf install iftop
G_MESSAGES_DEBUG='libdnf librepo libmodulemd' microdnf install iftop
G_MESSAGES_DEBUG=all microdnf install iftop
```

Reply:
Ok, actually you're right that if the context part used glog before, this shouldn't be unexpected and just add messages from the C++ part that were actually missing.

@rh-atomic-bot r+